### PR TITLE
Support batch input for median_filter2d

### PR DESCRIPTION
**Pull request by WindQAQ**
Addresses #230. Also meets the interface of mean_filter2d.

A general random test case is down below. Maybe there are some ways to speed up. It's now much more slower than scipy's implementation even for the input of shape [1, h, w, 1].
https://colab.research.google.com/drive/1UkNXxjTMHifKmHB2N5ojOHFojtXrOUIG

### CODE REVIEW

File: tensorflow_addons/image/filters.py [hunk 1]
```diff
@@ -142,80 +128,80 @@ def mean_filter2d(image,
...
+            x = tf.expand_dims(x, axis=-1)
+            patches = tf.image.extract_patches(
+                x,
+                sizes=[1, filter_shape[0], filter_shape[1], 1],
+                strides=[1, 1, 1, 1],
+                rates=[1, 1, 1, 1],
+                padding="VALID")
+            top = tf.nn.top_k(patches, k=ceil).values
+            median = (top[:, :, :, floor - 1] + top[:, :, :, ceil - 1]) / 2
+            return tf.dtypes.cast(median, x.dtype)
```
Comment (facaiy) [inline tensorflow_addons/image/filters.py]: Do we need to document the behavior? Say, return 6, rather than 6.5, for [5, 6, 7, 8]

**Comment by facaiy** [comment]
@Mainak431 Hi, would you mind taking a look?

File: tensorflow_addons/image/filters.py (hunk 1 above)
Comment (WindQAQ) [reply tensorflow_addons/image/filters.py]: Sure, it makes sense

**Comment by Mainak431** [comment]
@WindQAQ Thanks for the excellent contribution. Everything looks pretty fine.
Can you please add a small training example using this layer with a conv2d layer and dropout layer in tensorflow with maybe mnist dataset? So, that it can be directly used in the network itself?

**Comment by WindQAQ** [comment]
@Mainak431 Here is the example [colab notebook](https://colab.research.google.com/drive/1ZOAFTRdxKYDBpG75ZITukIr8MjUR2qDA). I will include this in addon's example in maybe the next PR.

**Comment by Mainak431** [comment]
Nice contribution.

**Comment by Squadrick** [comment]
@WindQAQ Thanks for the contribution.
